src/parquet/column/statistics.cc

```diff
@@ -64,27 +64,7 @@ TypedRowGroupStatistics<DType>::TypedRowGroupStatistics(const ColumnDescriptor*
 
   if (!encoded_min.empty()) { PlainDecode(encoded_min, &min_); }
   if (!encoded_max.empty()) { PlainDecode(encoded_max, &max_); }
-  has_min_max_ = !encoded_min.empty() && !encoded_max.empty();
-}
-
-template <>
-TypedRowGroupStatistics<ByteArrayType>::TypedRowGroupStatistics(
-    const ColumnDescriptor* schema,
-    const std::string& encoded_min, const std::string& encoded_max, int64_t num_values,
-    int64_t null_count, int64_t distinct_count, MemoryAllocator* allocator)
-    : allocator_(allocator), min_buffer_(0, allocator_), max_buffer_(0, allocator_) {
-  IncrementNumValues(num_values);
-  IncrementNullCount(null_count);
-  IncrementDistinctCount(distinct_count);
-
-  SetDescr(schema);
-
-  min_.len = encoded_min.size();
-  min_.ptr = reinterpret_cast<const uint8_t*>(encoded_min.c_str());
-  max_.len = encoded_max.size();
-  max_.ptr = reinterpret_cast<const uint8_t*>(encoded_max.c_str());
-
-  has_min_max_ = !encoded_min.empty() && !encoded_max.empty();
+  has_min_max_ = !encoded_min.empty() || !encoded_max.empty();
```

Comment:
The call in `reader-internal.cc`is actually unused, I'd put it there before Metadata class appeared, it can be safely removed now. Regarding `num_values`, see #145.

`Statistics` should be converted to `EncodedStatistics` because there's a policy not to expose Thrift-generated classes to the public interface.